### PR TITLE
docs: Self-hosted experimental

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds documentation for experimental options

## Context:

See #9152.

At this point I don't want to rename them, just document them.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-experimental.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## RENOVATE_PAGINATE_ALL
+
+If set to any value, Renovate will always paginate requests to GitHub fully, instead of stopping after 10 pages.
+
+## RENOVATE_REUSE_PACKAGE_LOCK
+
+If set to "false" (string), Renovate will remove any existing `package-lock.json` before attempting to update it.
+
+## RENOVATE_USER_AGENT
```
Comment (viceice) [inline docs/usage/self-hosted-experimental.md:39]: Should be converted to an admin function? But i never used that.

File: docs/usage/self-hosted-experimental.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Experimental options which are commonly used and for which there is no external solution in sight should be replaced with official options.
+
+Please use experimental options at your own risk - they may be removed or behavior changed in any release, not just major, however all attempts will be made not to inconvenience anyone.
+
+## GITLAB_IGNORE_REPO_URL
+
+If set to any value, Renovate will ignore the Project's `http_url_to_repo` value and instead construct the git URL manually.
+
+## RENOVATE_CACHE_NPM_MINUTES
```
Comment (viceice) [inline docs/usage/self-hosted-experimental.md:17]: We should have an admin option for all datasources. Currently  every datasource has it's own value 😕

File: docs/usage/self-hosted-configuration.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ description: Self-Hosted Configuration usable in renovate.json or package.json
 
 The below configuration options are applicable only if you are running your own instance ("bot") of Renovate.
 
+Please also see [Self-Hosted Experimental Options](./self-hosted-experimental.md).
```
Comment (viceice) [inline docs/usage/self-hosted-configuration.md:10]: Should we add this to the sidebar too ?

**Review by viceice — APPROVED** [review]

File: docs/usage/self-hosted-configuration.md (hunk 3 above)
Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:10]: I thought best not to. We already have a lot, and the whole idea of them is to keep them a little out of sight :)

File: docs/usage/self-hosted-experimental.md (hunk 2 above)
Comment (rarkins) [reply docs/usage/self-hosted-experimental.md:17]: Yes agreed, it should be configurable and then this option not necessary.

File: docs/usage/self-hosted-experimental.md (hunk 1 above)
Comment (rarkins) [reply docs/usage/self-hosted-experimental.md:39]: Agreed

**Pull request merged**